osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/configuration/service/SetConfigurationObjectService.java

```diff
@@ -0,0 +1,125 @@
+        this.dlmsHelper = dlmsHelper;
+    }
+
+    public AccessResultCode setConfigurationObject(final DlmsConnectionManager conn,
+            final ConfigurationObjectDto configurationToSet, final ConfigurationObjectDto configurationOnDevice)
+            throws ProtocolAdapterException {
+
+        final DataObject dataObject = this.buildSetParameterData(configurationToSet, configurationOnDevice);
+        LOGGER.info("ConfigurationObject SetParameter Data : {}", this.dlmsHelper.getDebugInfo(dataObject));
+
+        final AttributeAddress attributeAddress = AttributeAddressFactory.getConfigurationObjectAddress();
+        final SetParameter setParameter = new SetParameter(attributeAddress, dataObject);
+        conn.getDlmsMessageListener().setDescription(
+                "SetConfigurationObject AttributeAddress: " + JdlmsObjectToStringUtil.describeAttributes(
+                        attributeAddress));
+        return this.getAccessResultCode(conn, setParameter);
+    }
+
+    private AccessResultCode getAccessResultCode(final DlmsConnectionManager conn, final SetParameter setParameter) {
+        LOGGER.info("Set ConfigurationObject using SetParameter {}", setParameter);
+        try {
+            return conn.getConnection().set(setParameter);
+        } catch (final IOException e) {
+            throw new ConnectionException(e);
+        }
+    }
+
+    abstract DataObject buildSetParameterData(ConfigurationObjectDto configurationToSet,
+            ConfigurationObjectDto configurationOnDevice) throws ProtocolAdapterException;
+
+    BitString getFlags(final ConfigurationObjectDto configurationToSet,
+            final ConfigurationObjectDto configurationOnDevice) throws ProtocolAdapterException {
+        final List<ConfigurationFlagDto> flagsToSet = new ArrayList<>();
+        this.addSettableFlags(configurationToSet, flagsToSet);
+        this.addDeviceFlags(configurationOnDevice, flagsToSet);
+        final byte[] flagBytesToSet = this.toBytes(flagsToSet);
+        return new BitString(flagBytesToSet, NUMBER_OF_FLAG_BITS);
+    }
+
+    private void addSettableFlags(final ConfigurationObjectDto configurationToSet,
+            final List<ConfigurationFlagDto> flagsToSet) {
+        configurationToSet.getConfigurationFlags().getFlags().stream().filter(
+                flagToSet -> !flagToSet.getConfigurationFlagType().isReadOnly()).forEach(flagsToSet::add);
+    }
+
+    // Fill missing flagsToSet with flags from configurationOnDevice
+    private void addDeviceFlags(final ConfigurationObjectDto configurationOnDevice,
+            final List<ConfigurationFlagDto> flagsToSet) {
+        configurationOnDevice.getConfigurationFlags().getFlags().forEach(flagOnDevice -> {
+            if (flagsToSet.stream().noneMatch(
+                    flagToSet -> flagToSet.getConfigurationFlagType() == flagOnDevice.getConfigurationFlagType())) {
+                flagsToSet.add(flagOnDevice);
```

Comment:
consider changing to List.addAll with filter to simplify this statement.

Reply:
I'm not aware of a List.addAll with filter method in Java. Could you provide a code snippet to show what you mean?

Reply:
@coendamen could you give an example of what you mean?